### PR TITLE
Making single port 8000 default and update NetworkErrorBoundary

### PR DESCRIPTION
**Pull request by mattdepaula**
## What
It solves #3647 

## How
I just followed the tasks described

- [x] set API_URL=/api/v1/ in the default .env files and kube manifests
- [x]  set 502 response codes as a reason for showing the network boundary error
- [x] update docs

## Pre-merge Checklist
- [ x] *Run integration tests*
- [ ] *Publish Docker images*

## Recommended reading order
1. `.env`
2. `.env.dev`
3. `kube/overlays/stable/.env`
4. `kube/overlays/dev/.env`
5. `NetworkErrorBoundary.tsx`

### CODE REVIEW

File: airbyte-webapp/src/components/NetworkErrorBoundary/NetworkErrorBoundary.tsx
Comment (jrhizor) [inline airbyte-webapp/src/components/NetworkErrorBoundary/NetworkErrorBoundary.tsx:17]: I thought this section needed to read more like:

```
  static getDerivedStateFromError(error: {
    message: string;
    status?: number;
  }): { unReachServer: boolean } {
    // Update state so the next render will show the fallback UI.
    return {
      unReachServer:
        error.message === "Failed to fetch" ||
        (error.status !== undefined && error.status === 502),
    };
  }
```

It's trying to set the `unReachServer` boolean to whether or not it should show the "server is loading error". We want to show that error when there's an error message that contains "Failed to fetch" or if the status code is set and the status code is 502 (which means the underlying service we're proxying through isn't available yet).

However, both versions (the one in this PR and the version above), I got a `Uncaught SyntaxError: Unexpected token < in JSON at position 0`. I believe this is because `/api/v1/workspaces/get` is getting an HTML `An error occurred` 502 page from nginx, which is not JSON.

@Jamakase what's the best way to handle this?

**Review by jrhizor — CHANGES_REQUESTED** [review]
Wow this was super fast! Thanks!

Two things:
1. Can you also modify the port forwarding instructions in `docs/deploying-airbyte/*`?
2. Unfortunately we're blocked on merging this until the error handling is fixed. Waiting to see what's necessary from Artem.

File: airbyte-webapp/src/components/NetworkErrorBoundary/NetworkErrorBoundary.tsx
Comment (Jamakase) [reply airbyte-webapp/src/components/NetworkErrorBoundary/NetworkErrorBoundary.tsx:17]: In PR #3438 I totally refactored how we handle errors and introduced a richer error hierarchy. You could merge it as is and I will update it to the same format.

It also makes parsing error's response optional, so it won't fail if the error returned is not json.

Comment (jrhizor) [reply airbyte-webapp/src/components/NetworkErrorBoundary/NetworkErrorBoundary.tsx:17]: Oh that's great.

I think that since this won't actually show the "server is loading" error until  #3438 it probably makes sense to wait until  #3438  is merged, merge the changes into this PR and then merge the PR? Otherwise first time users will be confused about the behavior like before we added the "server is loading" error in the first place.

Comment (Jamakase) [reply airbyte-webapp/src/components/NetworkErrorBoundary/NetworkErrorBoundary.tsx:17]: @jrhizor sounds reasonable to me. I need to address few comments for that PR and after that we could merge. Should I merge master into this branch as soon as we are ready for merging?

Comment (jrhizor) [reply airbyte-webapp/src/components/NetworkErrorBoundary/NetworkErrorBoundary.tsx:17]: Yep, that'd be perfect.

**Comment by MatheusdiPaula** [comment]
> Wow this was super fast! Thanks!
> 
> Two things:
> 
> 1. Can you also modify the port forwarding instructions in `docs/deploying-airbyte/*`?
> 2. Unfortunately we're blocked on merging this until the error handling is fixed. Waiting to see what's necessary from Artem.

1. Done!

**Comment by mjmanas0699** [comment]
working on load balancer as well in k8s
**API_URL=/api/v1/**

Thanks! @MatheusdiPaula

**Comment by marcosmarxm** [comment]
@MatheusdiPaula could you update your branch and resolve the conflicts?

**Comment by MatheusdiPaula** [comment]
> @MatheusdiPaula could you update your branch and resolve the conflicts?

Branch updated!

**Comment by jrhizor** [comment]
Thanks so much for this @MatheusdiPaula!

I'm going to merge a followup after this but this looks good to go!

**Review by jrhizor — APPROVED** [review]

**Pull request merged**
